csvcubed/csvcubed/cli/inspect/inspectdatasetmanager.py

```diff
@@ -0,0 +1,188 @@
+        _logger.info("Successfully loaded csv into dataframe.")
+
+        return dataset
+    except Exception as ex:
+        raise CsvToDataFrameLoadFailedException from ex
+
+
+def get_measure_col_name_from_dsd(
+    components: List[QubeComponentResult],
+) -> Optional[str]:
+    """
+    Identifies the name of measure column.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `Optional[str]`
+    """
+    filtered_components = _filter_components_from_dsd(
+        components,
+        ComponentField.Property,
+        ComponentPropertyAttributeURI.MeasureType.value,
+    )
+
+    return (
+        filtered_components[0].csv_col_title
+        if filtered_components[0] and filtered_components[0].csv_col_title != ""
+        else None
+    )
+
+
+def get_unit_col_name_from_dsd(
+    components: List[QubeComponentResult],
+) -> Optional[str]:
+    """
+    Identifies the name of unit column.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `Optional[str]`
+    """
+    filtered_components = _filter_components_from_dsd(
+        components,
+        ComponentField.Property,
+        ComponentPropertyAttributeURI.UnitMeasure.value,
+    )
+
+    return (
+        filtered_components[0].csv_col_title
+        if filtered_components[0] and filtered_components[0].csv_col_title != ""
+        else None
+    )
+
+
+def get_single_measure_from_dsd(
+    components: List[QubeComponentResult], json_path: Path
+) -> DatasetSingleMeasureResult:
+    """
+    Identifies the measure uri and label for single-measure dataset.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DatasetSingleMeasureResult`
+    """
+    filtered_components = _filter_components_from_dsd(
+        components, ComponentField.PropertyType, ComponentPropertyType.Measure.value
+    )
+
+    if len(filtered_components) != 1:
+        raise InvalidNumberOfRecordsException(
+            excepted_num_of_records=1, num_of_records=len(filtered_components)
+        )
+
+    return DatasetSingleMeasureResult(
+        measure_uri=get_component_property_as_relative_path(
+            json_path, filtered_components[0].property
+        ),
+        measure_label=filtered_components[0].property_label,
+    )
+
+
+def get_dataset_observations_info(
+    dataset: pd.DataFrame,
+) -> DatasetObservationsInfoResult:
+    """
+    Generates the `DatasetObservationsInfoResult` from the dataset.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DatasetObservationsInfoResult`
+    """
+    return DatasetObservationsInfoResult(
+        len(dataset.index),
+        dataset.duplicated().sum(),
+        dataset.head(n=10),
+        dataset.tail(n=10),
+    )
+
+
+def get_dataset_val_counts_info(
+    dataset: pd.DataFrame, measure_col: str, unit_col: str
+) -> DatasetObservationsByMeasureUnitInfoResult:
+    """
+    Generates the `DatasetObservationsByMeasureUnitInfoResult` from the dataset.
```

Comment:
You don't  have to do this but since you seem to like using the magic keywords, if you do :class:`~full.path.to.DatasetObservationsByMeasureUnitInfoResult` then sphinx will create the linking for you in the [API Docs](https://gss-cogs.github.io/csvcubed-docs/api-docs/csvcubed/csvcubed.cli.inspect.html#metadata-processor).

Reply:
@robons Thanks for the info. I will keep a note of this and will update throughout as part of the next PR.